core/src/main/java/org/springframework/ldap/odm/core/impl/ObjectMetaData.java

```diff
@@ -85,29 +87,33 @@ public AttributeMetaData getAttribute(Field field) {
         return fieldToAttribute.get(field);
     }
     
-    public ObjectMetaData(Class<?> clazz) {
+    public ObjectMetaData(final Class<?> clazz) {
         if (LOG.isDebugEnabled()) {
             LOG.debug(String.format("Extracting metadata from %1$s", clazz));
         }
         
         // Get object class metadata - the @Entity annotation
-        Entry entity = clazz.getAnnotation(Entry.class);
-        if (entity != null) {
+        // findAllMergedAnnotations will return set of inherited annotations and apply them from superclass down to subclass
+        Set<Entry> entities = AnnotatedElementUtils.findAllMergedAnnotations(clazz,Entry.class);
```

Comment:
While I agree that we want to resolve inheritance and meta-annotation concerns, I don't think we want to merge multiple `Entry` annotations. The reason is that each `Entry` can come with a `base` and it's not clear which should be picked.

Instead, it may be valuable to look at [Spring Security's `AuthorizationAnnotationUtil#findUniqueAnnotation`](https://github.com/spring-projects/spring-security/blob/main/core/src/main/java/org/springframework/security/authorization/method/AuthorizationAnnotationUtils.java) for some inspiration on how to resolve the `@Entry` annotation while still erroring if the result is ambiguous.

Reply:
There is a use case where say you have a inetOrgPerson class which inherits from a organizationalPerson class which inherits from a person class. If you want to reference inetOrgPersons in one query, but person in another they could have different Entry annotation values. By enumerating through all annotations from the top of the class tree and overriding with subclasses it still allows this behavior.

Reply:
I understand the use case for filters, but the `base` value remains ambiguous. Which `base` should you pick?

Reply:
Sorry for the long absence. I would argue that the default should be to take the lowest class in the class inheritance tree (closest to the instantiated class type) with the ability for the class writer to disambiguate by defining the base on the subclass.

Reply:
I don't want to get hung up on this feature, and it merits further discussion. Could we create a separate ticket to discuss how `objectClasses` inheritance should work? I'm happy to add to that ticket my thoughts and we can continue the conversation over there.

For this ticket, can you please leave inheritance as follows:

1. The target class must have an `@Entry` annotation or meta-annotation
2. No parent `@Entry` annotations are considered
3. All reachable fields are collected (what you are already doing [here](https://github.com/spring-projects/spring-ldap/pull/591/files#r788779726))

The reason for keeping 3 is that it's clear from the ticket that not collecting all reachable fields is the thing that is primarily surprising to folks.

I'm also happy to update the PR with a polish that does the above if you aren't available. If I don't hear from you within a month or so, I'll proceed as I've described so that it can land in time for the 3.3.0 release.